### PR TITLE
Call SecTrustEvaluateAsync from correct queue.

### PR DESCRIPTION
**Pull request by Lukasa**
Motivation:

SecTrustEvaluateAsync documents that it must be called from the same
queue it will call you back on. We failed to do this. Let's fix that up.
While we're here, we should use the non-deprecated API where it's
practical to do so.

Modifications:

- Async out to the queue before calling SecTrustEvaluateAsync.
- Call SecTrustEvaluateAsyncWithError where it's available.

Result:

Following the rules.

### CODE REVIEW

**Review by glbrntt — APPROVED** [review]
LGTM

File: Sources/NIOSSL/SecurityFrameworkCertificateVerification.swift [hunk 1]
```diff
@@ -55,24 +55,42 @@ extension SSLConnection {
...
-                    promise.succeed(.certificateVerified)
-                default:
-                    // Oops, we failed.
-                    promise.succeed(.failed)
+            // SecTrustEvaluateAsync and its cousin withError require that they are called from the same queue given to
+            // them as a parameter. Thus, we async away now.
+            callbackQueue.async {
+                let result: OSStatus
+
+                if #available(iOS 13, macOS 10.15, macCatalyst 13, tvOS 13, watchOS 6, *) {
```
Comment (glbrntt) [inline Sources/NIOSSL/SecurityFrameworkCertificateVerification.swift:63]: CI isn't happy about this:

```
SecurityFrameworkCertificateVerification.swift:63:66: error: unrecognized platform name 'macCatalyst'
```

**Review by glbrntt — APPROVED** [review]

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Comment by Lukasa** [comment]
Merging over the nightly failure.

**Pull request merged**